syllabus/curricula/plr/task_sampler.py

```diff
@@ -285,16 +377,64 @@ def sample(self, strategy=None):
                 if np.random.rand() > self.nu or not proportion_seen < 1.0:
                     return self._sample_replay_level()
 
-            # Otherwise, sample a new level
-            return self._sample_unseen_level()
+            # Otherwise, evaluate a new level
+            if self.robust_plr:
+                self.update_with_episode_data(self._evaluate_unseen_level())
+                return self.sample(strategy=strategy)
+            else:
+                # Otherwise, sample a new level
+                return self._sample_unseen_level()
+
```

Comment:
This is a good start, but it's going to be very inefficient to do these evaluations in the main process. We'll probably want to batch and multiprocess them in the future, but for now this is good as a proof of concept.